src/app/chat/view.nim

```diff
@@ -422,6 +424,21 @@ QtObject:
 
   proc loadingMessagesChanged*(self: ChatsView, value: bool) {.signal.}
 
+  proc asyncMessageLoad*(self: ChatsView, chatId: string) {.slot.} =
+    spawnAndSend(self, "asyncMessageLoaded") do: # Call self.ensResolved(string) when ens is resolved
+      $(%*{
+        "chatId": chatId,
+        "messages": callPrivateRPC("chatMessages".prefix, %* [chatId, "", 20]).parseJson()["result"],
+        "reactions": callPrivateRPC("emojiReactionsByChatID".prefix, %* [chatId, "", 20]).parseJson()["result"]
```

Comment:
Could we instead call the `libstatus/chat` functions. Calling `callPrivateRPC` in a view seems a bit too close to the implementation. Also, it would enable us to add error handling with a `success` var

Reply:
That sounds good. I'll do that